third_party/libusb/webport/src/libusb_js_proxy_unittest.cc

```diff
@@ -616,6 +716,75 @@ TEST_P(LibusbJsProxyWithDeviceTest, DeviceResettingFailure) {
   EXPECT_EQ(libusb()->LibusbResetDevice(device_handle_), LIBUSB_ERROR_OTHER);
 }
 
+// Test `LibusbClaimInterface()` successful scenario.
+TEST_P(LibusbJsProxyWithDeviceTest, InterfaceClaiming) {
+  constexpr int kInterfaceNumber = 12;
+  // Arrange.
+  global_context()->WillReplyToRequestWith("libusb", "claimInterface",
+                                           /*arguments=*/
+                                           ArrayValueBuilder()
+                                               .Add(kJsDeviceId)
+                                               .Add(kJsDeviceHandle)
+                                               .Add(kInterfaceNumber)
+                                               .Get(),
+                                           /*result_to_reply_with=*/Value());
+
+  // Act.
+  EXPECT_EQ(libusb()->LibusbClaimInterface(device_handle_, kInterfaceNumber),
+            LIBUSB_SUCCESS);
+}
+
+// Test `LibusbClaimInterface()` failure due to the JS call returning an error.
+TEST_P(LibusbJsProxyWithDeviceTest, InterfaceClaimingFailure) {
+  constexpr int kInterfaceNumber = 12;
+  // Arrange.
+  global_context()->WillReplyToRequestWithError(
+      "libusb", "claimInterface",
+      /*arguments=*/
+      ArrayValueBuilder()
+          .Add(kJsDeviceId)
+          .Add(kJsDeviceHandle)
+          .Add(kInterfaceNumber)
+          .Get(),
+      /*error_to_reply_with=*/"fake error");
+
+  // Act.
+  EXPECT_EQ(libusb()->LibusbClaimInterface(device_handle_, kInterfaceNumber),
+            LIBUSB_ERROR_OTHER);
+}
+
+// Test that `LibusbGetBusNumber()` returns an incremented value if the
+// `LibusbClaimInterface()` method failed previously.
+TEST_P(LibusbJsProxyWithDeviceTest,
+       BusNumberWorkaroundAfterInterfaceClaimingFailure) {
+  constexpr int kInterfaceNumber = 12;
+  // Arrange.
+  global_context()->WillReplyToRequestWith(
+      "libusb", "listDevices",
+      /*arguments=*/Value(Value::Type::kArray), MakeListDevicesFakeJsReply());
+  global_context()->WillReplyToRequestWithError(
+      "libusb", "claimInterface",
+      /*arguments=*/
+      ArrayValueBuilder()
+          .Add(kJsDeviceId)
+          .Add(kJsDeviceHandle)
+          .Add(kInterfaceNumber)
+          .Get(),
+      /*error_to_reply_with=*/"fake error");
+
+  // Act.
+  EXPECT_EQ(libusb()->LibusbClaimInterface(device_handle_, kInterfaceNumber),
+            LIBUSB_ERROR_OTHER);
+  // Obtain the list of devices again.
+  std::vector<libusb_device*> devices = GetLibusbDevices();
+  ASSERT_EQ(devices.size(), 1U);
+
+  // Assert. The bus number should be incremented now.
+  EXPECT_EQ(libusb()->LibusbGetBusNumber(devices[0]), 2);
```

Comment:
nit: I would add an assertion before the call that the BusNumber is 1, to illustrate the assertion